src/docfx/build/toc/TableOfContentsMap.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace Microsoft.Docs.Build
+{
+    /// <summary>
+    /// The mappings between toc and document
+    /// </summary>
+    internal class TableOfContentsMap
+    {
+        private readonly IEnumerable<Document> _tocs;
+
+        private readonly Dictionary<Document, HashSet<Document>> _documentToTocs;
+
+        private readonly ConcurrentDictionary<Document, Document> _nearestTocs = new ConcurrentDictionary<Document, Document>();
+
+        public TableOfContentsMap(IEnumerable<Document> tocs, Dictionary<Document, HashSet<Document>> documentToTocs)
```

Comment:
Suggest to use `List<Document> tocs` over `IEnumerable<Document>`.

Reply:
suggest should follow a real reason, right?

---
In reply to: [184584544](https://github.com/dotnet/docfx/pull/2696#discussion_r184584544) [](ancestors = 184584544)